public/js/components/Editor.js

```diff
@@ -141,6 +142,22 @@ const Editor = React.createClass({
     this.editor.setText(text);
   },
 
+  setMode(sourceText) {
+    const contentType = sourceText.get("contentType");
+
+    if (contentType.includes("javascript")) {
+      this.editor.setMode({ name: "javascript" });
+    } else if (contentType === "text/wasm") {
+      this.editor.setMode({ name: "wasm" });
+    } else if (sourceText.get("text").match(/^\s*</)) {
+      // Use HTML mode for files in which the first non whitespace
+      // character is `<` regardless of extension.
+      this.editor.setMode({ name: "htmlmixed" });
+    } else {
+      this.editor.setMode({ name: "text" });
+    }
+  },
```

Comment:
The `Editor` component is the main thing that should do this work, I don't like have 2 separate modules that split the work for no great reason. In this case, yeah, we're forced to have source-editor.js for compatibility. 

You're right that we haven't ordered this correctly; the lifecycle methods should be grouped together. I'll move `componentDidMount` down and then it's utils, lifecycle, render, in that order. There's an eslint rule to enforce this that I thought got turned on in m-c...